### PR TITLE
Fix layout direction override (PWENG-39)

### PR DESCRIPTION
**Pull request by alexrepty**
### Checklist
- [X] If applicable, unit tests
- [ ] If applicable, create follow-up issues for `purchases-android` and hybrids

### Motivation
When `Purchases.shared.overridePreferredUILocale(_:)` is set to an RTL locale (e.g. Hebrew or Arabic) while the device system locale is LTR (e.g. English), the paywall renders translated strings correctly but the layout direction stays LTR. Close buttons, package cards, and text alignment all remain left-anchored, producing a broken RTL experience.

Resolves PWENG-39

### Description
SwiftUI's layout engine derives text direction and `HStack` alignment from the `\.locale` environment value. When that key is not explicitly set, SwiftUI inherits it from the host app's environment, which reflects the system locale — not the locale override. Because `overridePreferredUILocale` was only threaded through the content-selection path (string bundle lookup), the environment was never updated and layout stayed LTR.

**Fix**: Inject `\.locale` and `\.layoutDirection` into the SwiftUI environment at the point where the resolved locale is already available, in both the V1 (`LoadedOfferingPaywallView`) and V2 (`PaywallsV2View`) render paths. A `Locale.swiftUILayoutDirection` helper maps `Locale.characterDirection(forLanguage:)` to `SwiftUI.LayoutDirection`.

### Testing
Added `LocaleLayoutDirectionTests` unit tests covering RTL languages (Hebrew, Arabic, Farsi, Urdu) and LTR languages (English, French, Japanese), including region-qualified identifiers and the empty-string edge case.
Expanded `PaywallViewLocalizationTests` with Hebrew (he-IL) and Arabic (ar) snapshot regression tests. Each test uses a locale-specific offering with native translations (just AI-translated, since it won't actually be facing end users) and renders with `localeOverride` set, so a future regression (RTL strings but LTR layout) would produce a diff in the reference PNGs.

---

> [!NOTE]
> **Medium Risk**
> Changes SwiftUI environment injection for `locale`/`layoutDirection` in both V1 and V2 paywall render paths, which can affect layout across all paywalls and locales. Risk is mitigated by added unit and snapshot coverage for RTL and locale-resolution edge cases.
> 
> **Overview**
> Fixes RTL rendering when `overridePreferredUILocale` is set by explicitly injecting the resolved content `Locale` and matching `LayoutDirection` into the SwiftUI environment for both V1 (`LoadedOfferingPaywallView`) and V2 (`PaywallsV2View`) paywalls.
> 
> Adds `Locale.swiftUILayoutDirection` plus a shared `Locale.selectPreferredLocale` helper (used by V2 locale resolution), and expands test coverage with new RTL/LTR unit tests and Hebrew/Arabic snapshot regressions (plus a small snapshot-recording helper in `TestCase`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5e6b9bcd347502803cab4009cfc96cfe6dcf6b9e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: RevenueCatUI/Templates/V2/PaywallsV2View.swift [hunk 1]
```diff
@@ -165,13 +165,16 @@ struct PaywallsV2View: View {
...
+        let resolvedLocale = self.purchaseHandler.preferredLocaleOverride ?? .current
+        return LoadedPaywallsV2View(
             introOfferEligibilityContext: introOfferEligibilityContext,
             paywallState: paywallState,
             uiConfigProvider: self.uiConfigProvider,
             selectedPackageContext: self.selectedPackageContext,
             onDismiss: self.onDismiss
         )
+        .environment(\.locale, resolvedLocale)
+        .environment(\.layoutDirection, resolvedLocale.swiftUILayoutDirection)
```
Comment (alexrepty) [inline RevenueCatUI/Templates/V2/PaywallsV2View.swift]: This is the actual meat of the change. Setting the environment variables will correctly apply the locale and layout direction.

**Review by JZDesign — APPROVED** [review]

File: RevenueCatUI/PaywallView.swift [hunk 2]
```diff
@@ -544,6 +544,8 @@ struct LoadedOfferingPaywallView: View {
             showZeroDecimalPlacePrices: self.showZeroDecimalPlacePrices
         )
         let view = paywallView(withConfig: configuration)
+            .environment(\.locale, self.locale)
```
Comment (MonikaMateska) [inline RevenueCatUI/PaywallView.swift:547]: Is there any reason why we do not use the resolved locale here?

Comment (alexrepty) [reply RevenueCatUI/PaywallView.swift:547]: The stored property is resolved when it's computed, so it ends up working correctly with this change, i.e.:
Image
